lightning/src/ln/channelmanager.rs

```diff
@@ -206,6 +206,24 @@ impl Readable for PaymentId {
 		Ok(PaymentId(buf))
 	}
 }
+
+/// An identifier used to uniquely identify an intercepted HTLC to LDK.
+/// (C-not exported) as we just use [u8; 32] directly
```

Comment:
I'm a bit confused how an attacker would get the `incoming_shared_secret` without having the node's privkey? I agree that if the attacker had that piece of info + an intercept scid, they may be able to probe for pending intercepts via some timing attack, though note that the error returned is identical to the "don't have available channel for forwarding" error.

Reply:
I agree this would assume either a leak of the sender secret key or the routing hop secret key. In both cases, I think we can assume you're opening the door to few privacy leaks and worst. As the `InterceptId` is based on the shared secret, an entity with the leak of the intercept id shouldn't be able to hit the `Occupied` entry L5256 in `channelmanager`.